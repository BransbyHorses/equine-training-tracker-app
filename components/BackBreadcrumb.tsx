import React from "react";

import { useRouter } from "next/router";

import { Breadcrumbs, Link } from "@mui/material";
import ArrowLeftIcon from "@mui/icons-material/ArrowLeft";

const BackBreadcrumb = () => {
	const router = useRouter();
	return (
		<Breadcrumbs>
			<Link
				underline="hover"
				color="inherit"
				sx={{ display: "flex", alignItems: "center", cursor: "pointer" }}
				onClick={() => router.back()}
			>
				<ArrowLeftIcon /> Back
			</Link>
		</Breadcrumbs>
	);
};

export default BackBreadcrumb;
