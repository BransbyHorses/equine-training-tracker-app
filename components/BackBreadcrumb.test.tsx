import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import BackBreadcrumb from "./BackBreadcrumb";

describe("BackBreadcrumb", () => {
	it("renders with all props", () => {
		const tree = renderer.create(<BackBreadcrumb link="/" />);
		expect(tree).toMatchSnapshot();
	});
});
