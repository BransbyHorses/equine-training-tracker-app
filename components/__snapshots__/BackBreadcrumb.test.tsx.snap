// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackBreadcrumb renders with all props 1`] = `
<nav
  className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-7a84y9-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      className="MuiBreadcrumbs-li"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1qt0ygz-MuiTypography-root-MuiLink-root"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m14 7-5 5 5 5V7z"
          />
        </svg>
         Back
      </a>
    </li>
  </ol>
</nav>
`;
